enterprise/ssl-impl/src/test/java/io/crate/protocols/mqtt/CrateMqttTransportIntegrationTest.java

```diff
@@ -0,0 +1,135 @@
+import org.elasticsearch.test.ESIntegTestCase;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import javax.net.ssl.SSLSocketFactory;
+import java.io.File;
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+import static io.crate.protocols.ssl.SslConfigurationTest.getAbsoluteFilePathFromClassPath;
+import static org.hamcrest.Matchers.is;
+
+
+@ESIntegTestCase.ClusterScope(numDataNodes = 1, numClientNodes = 0, supportsDedicatedMasters = false)
+public class CrateMqttTransportIntegrationTest extends SQLTransportIntegrationTest {
+
+    private static File trustStoreFile;
+    private static File keyStoreFile;
+    private MqttClient client;
+
+    @BeforeClass
+    public static void beforeIntegrationTest() throws IOException {
+        keyStoreFile = getAbsoluteFilePathFromClassPath("keystore.jks");
+        trustStoreFile = getAbsoluteFilePathFromClassPath("truststore.jks");
+        System.setProperty("javax.net.ssl.trustStore", trustStoreFile.getAbsolutePath());
+        System.setProperty("javax.net.ssl.trustStorePassword", "truststorePassword");
+    }
+
+    @AfterClass
+    public static void afterIntegrationTest() {
+        System.clearProperty("javax.net.ssl.trustStore");
+        System.clearProperty("javax.net.ssl.trustStorePassword");
+    }
+
+    @Override
+    protected Settings nodeSettings(int nodeOrdinal) {
+        return Settings.builder()
+            .put(super.nodeSettings(nodeOrdinal))
+            .put(SharedSettings.ENTERPRISE_LICENSE_SETTING.getKey(), true)
+            .put(Netty4MqttServerTransport.MQTT_ENABLED_SETTING.getKey(), true)
+            .put(Netty4MqttServerTransport.MQTT_PORT_SETTING.getKey(), 8883)
+            .put(SslConfigSettings.SSL_MQTT_ENABLED.getKey(), true)
+            .put(SslConfigSettings.SSL_KEYSTORE_FILEPATH.getKey(), keyStoreFile)
+            .put(SslConfigSettings.SSL_KEYSTORE_PASSWORD.getKey(), "keystorePassword")
+            .put(SslConfigSettings.SSL_KEYSTORE_KEY_PASSWORD.getKey(), "serverKeyPassword")
+            .put(SslConfigSettings.SSL_TRUSTSTORE_FILEPATH.getKey(), trustStoreFile)
+            .put(SslConfigSettings.SSL_TRUSTSTORE_PASSWORD.getKey(), "truststorePassword")
+            .build();
+    }
+
+    @Before
+    public void setupMqttTableAndIngestRule() {
+        execute("CREATE TABLE IF NOT EXISTS mqtt.raw (\n" +
+                "  client_id STRING,\n" +
+                "  packet_id INTEGER,\n" +
+                "  topic STRING,\n" +
+                "  ts TIMESTAMP,\n" +
+                "  payload OBJECT(DYNAMIC),\n" +
+                "  PRIMARY KEY (\"client_id\", \"packet_id\")\n" +
+                ") WITH (column_policy = 'strict', number_of_replicas = '0')");
+
+        execute("CREATE INGEST RULE ingest_topic_t1 ON " + MqttIngestService.SOURCE_IDENT +
+                " WHERE topic = ? INTO mqtt.raw", new Object[]{"t1"});
```

Comment:
Do we need all that to test an ssl connection?

Reply:
The mqtt client doesn't offer any way to wait for a PUBACK or reply for a publish. 
Do you have any idea on how else can we test that?